File: ios/Models/ProcessorIDScanResultsSoFar.swift
import Foundation

public class ProcessorIDScanResultsSoFar: JSONParams {
  private var object: [AnyHashable: Any?]

  override init(target: [AnyHashable: Any]?) {
    self.object = [:]

    super.init(target: target)

    self.parseJSON()
  }

  private func parseJSON() {
    let photoIDNextStepEnumInt = self.getParam("photoIDNextStepEnumInt", type: Int.self)
    let fullIDStatusEnumInt = self.getParam("fullIDStatusEnumInt", type: Int.self)
    let faceOnDocumentStatusEnumInt = self.getParam("faceOnDocumentStatusEnumInt", type: Int.self)
    let textOnDocumentStatusEnumInt = self.getParam("textOnDocumentStatusEnumInt", type: Int.self)
    let expectedMediaStatusEnumInt = self.getParam("expectedMediaStatusEnumInt", type: Int.self)
    let unexpectedMediaEncounteredAtLeastOnce = self.getParam(
      "unexpectedMediaEncounteredAtLeastOnce", type: Bool.self)
    let documentData = self.getParam("documentData", type: String.self)
    let nfcStatusEnumInt = self.getParam("nfcStatusEnumInt", type: Int.self)
    let nfcAuthenticationStatusEnumInt = self.getParam(
      "nfcAuthenticationStatusEnumInt", type: Int.self)
    let barcodeStatusEnumInt = self.getParam("barcodeStatusEnumInt", type: Int.self)
    let mrzStatusEnumInt = self.getParam("mrzStatusEnumInt", type: Int.self)
    let idFoundWithoutQualityIssueDetected = self.getParam(
      "idFoundWithoutQualityIssueDetected", type: Bool.self)
    let idFacePhotoFoundWithoutQualityIssueDetected = self.getParam(
      "idFacePhotoFoundWithoutQualityIssueDetected", type: Bool.self)
    let idScanAgeV2GroupEnumInt = self.getParam("idScanAgeV2GroupEnumInt", type: Int.self)
    let didMatchIDScanToOCRTemplate = self.getParam("didMatchIDScanToOCRTemplate", type: Bool.self)
    let isUniversalIDMode = self.getParam("isUniversalIDMode", type: Bool.self)
    let matchLevel = self.getParam("matchLevel", type: Int.self)
    let matchLevelNFCToFaceMap = self.getParam("matchLevelNFCToFaceMap", type: Int.self)
    let faceMapAgeV2GroupEnumInt = self.getParam("faceMapAgeV2GroupEnumInt", type: Int.self)
    let watermarkAndHologramStatusEnumInt = self.getParam(
      "watermarkAndHologramStatusEnumInt", type: Int.self)

    self.object["documentData"] = documentData
    self.object["photoIDNextStepEnumInt"] = self.defaultIfNull(
      photoIDNextStepEnumInt, defaultValue: -1)
    self.object["fullIDStatusEnumInt"] = self.defaultIfNull(fullIDStatusEnumInt, defaultValue: -1)
    self.object["faceOnDocumentStatusEnumInt"] = self.defaultIfNull(
      faceOnDocumentStatusEnumInt, defaultValue: -1)
    self.object["textOnDocumentStatusEnumInt"] = self.defaultIfNull(
      textOnDocumentStatusEnumInt, defaultValue: -1)
    self.object["expectedMediaStatusEnumInt"] = self.defaultIfNull(
      expectedMediaStatusEnumInt, defaultValue: -1)
    self.object["unexpectedMediaEncounteredAtLeastOnce"] =
      unexpectedMediaEncounteredAtLeastOnce == true
    self.object["nfcStatusEnumInt"] = self.defaultIfNull(nfcStatusEnumInt, defaultValue: -1)
    self.object["nfcAuthenticationStatusEnumInt"] = self.defaultIfNull(
      nfcAuthenticationStatusEnumInt, defaultValue: -1)
    self.object["barcodeStatusEnumInt"] = self.defaultIfNull(barcodeStatusEnumInt, defaultValue: -1)
    self.object["mrzStatusEnumInt"] = self.defaultIfNull(mrzStatusEnumInt, defaultValue: -1)
    self.object["idFoundWithoutQualityIssueDetected"] = idFoundWithoutQualityIssueDetected == true
    self.object["idFacePhotoFoundWithoutQualityIssueDetected"] =
      idFacePhotoFoundWithoutQualityIssueDetected == true
    self.object["idScanAgeV2GroupEnumInt"] = self.defaultIfNull(
      idScanAgeV2GroupEnumInt, defaultValue: -1)
    self.object["didMatchIDScanToOCRTemplate"] = didMatchIDScanToOCRTemplate == true
    self.object["isUniversalIDMode"] = isUniversalIDMode == true
    self.object["matchLevel"] = self.defaultIfNull(matchLevel, defaultValue: -1)
    self.object["matchLevelNFCToFaceMap"] = self.defaultIfNull(
      matchLevelNFCToFaceMap, defaultValue: -1)
    self.object["faceMapAgeV2GroupEnumInt"] = self.defaultIfNull(
      faceMapAgeV2GroupEnumInt, defaultValue: -1)
    self.object["watermarkAndHologramStatusEnumInt"] = self.defaultIfNull(
      watermarkAndHologramStatusEnumInt, defaultValue: -1)
  }

  func getDict() -> NSDictionary {
    return NSDictionary(dictionary: self.object as [AnyHashable: Any])
  }
}

File: android/src/main/java/com/azifacemobile/models/ProcessorIDScanResultsSoFar.java
package com.azifacemobile.models;

import com.azifacemobile.models.abstracts.JSONParams;
import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;

import org.json.JSONObject;

public class ProcessorIDScanResultsSoFar extends JSONParams {
  private final WritableMap object;

  public ProcessorIDScanResultsSoFar(JSONObject target) {
    super(target);

    this.object = Arguments.createMap();

    this.parseJSON();
  }

  private void parseJSON() {
    Integer photoIDNextStepEnumInt = this.getParam( "photoIDNextStepEnumInt", Integer.class);
    Integer fullIDStatusEnumInt = this.getParam("fullIDStatusEnumInt", Integer.class);
    Integer faceOnDocumentStatusEnumInt = this.getParam("faceOnDocumentStatusEnumInt", Integer.class);
    Integer textOnDocumentStatusEnumInt = this.getParam("textOnDocumentStatusEnumInt", Integer.class);
    Integer expectedMediaStatusEnumInt = this.getParam("expectedMediaStatusEnumInt", Integer.class);
    Boolean unexpectedMediaEncounteredAtLeastOnce = this.getParam("unexpectedMediaEncounteredAtLeastOnce", Boolean.class);
    String documentData = this.getParam("documentData", String.class);
    Integer nfcStatusEnumInt = this.getParam("nfcStatusEnumInt", Integer.class);
    Integer nfcAuthenticationStatusEnumInt = this.getParam("nfcAuthenticationStatusEnumInt", Integer.class);
    Integer barcodeStatusEnumInt = this.getParam("barcodeStatusEnumInt", Integer.class);
    Integer mrzStatusEnumInt = this.getParam("mrzStatusEnumInt", Integer.class);
    Boolean idFoundWithoutQualityIssueDetected = this.getParam("idFoundWithoutQualityIssueDetected", Boolean.class);
    Boolean idFacePhotoFoundWithoutQualityIssueDetected = this.getParam("idFacePhotoFoundWithoutQualityIssueDetected", Boolean.class);
    Integer idScanAgeV2GroupEnumInt = this.getParam("idScanAgeV2GroupEnumInt", Integer.class);
    Boolean didMatchIDScanToOCRTemplate = this.getParam("didMatchIDScanToOCRTemplate", Boolean.class);
    Boolean isUniversalIDMode = this.getParam("isUniversalIDMode", Boolean.class);
    Integer matchLevel = this.getParam("matchLevel", Integer.class);
    Integer matchLevelNFCToFaceMap = this.getParam("matchLevelNFCToFaceMap", Integer.class);
    Integer faceMapAgeV2GroupEnumInt = this.getParam("faceMapAgeV2GroupEnumInt", Integer.class);
    Integer watermarkAndHologramStatusEnumInt = this.getParam("watermarkAndHologramStatusEnumInt", Integer.class);

    this.object.putString("documentData", documentData);
    this.object.putInt("photoIDNextStepEnumInt", this.defaultIfNull(photoIDNextStepEnumInt, -1));
    this.object.putInt("fullIDStatusEnumInt", this.defaultIfNull(fullIDStatusEnumInt, -1));
    this.object.putInt("faceOnDocumentStatusEnumInt", this.defaultIfNull(faceOnDocumentStatusEnumInt, -1));
    this.object.putInt("textOnDocumentStatusEnumInt", this.defaultIfNull(textOnDocumentStatusEnumInt, -1));
    this.object.putInt("expectedMediaStatusEnumInt", this.defaultIfNull(expectedMediaStatusEnumInt, -1));
    this.object.putBoolean("unexpectedMediaEncounteredAtLeastOnce", Boolean.TRUE.equals(unexpectedMediaEncounteredAtLeastOnce));
    this.object.putInt("nfcStatusEnumInt", this.defaultIfNull(nfcStatusEnumInt, -1));
    this.object.putInt("nfcAuthenticationStatusEnumInt", this.defaultIfNull(nfcAuthenticationStatusEnumInt, -1));
    this.object.putInt("barcodeStatusEnumInt", this.defaultIfNull(barcodeStatusEnumInt, -1));
    this.object.putInt("mrzStatusEnumInt", this.defaultIfNull(mrzStatusEnumInt, -1));
    this.object.putBoolean("idFoundWithoutQualityIssueDetected", Boolean.TRUE.equals(idFoundWithoutQualityIssueDetected));
    this.object.putBoolean("idFacePhotoFoundWithoutQualityIssueDetected", Boolean.TRUE.equals(idFacePhotoFoundWithoutQualityIssueDetected));
    this.object.putInt("idScanAgeV2GroupEnumInt", this.defaultIfNull(idScanAgeV2GroupEnumInt, -1));
    this.object.putBoolean("didMatchIDScanToOCRTemplate", Boolean.TRUE.equals(didMatchIDScanToOCRTemplate));
    this.object.putBoolean("isUniversalIDMode", Boolean.TRUE.equals(isUniversalIDMode));
    this.object.putInt("matchLevel", this.defaultIfNull(matchLevel, -1));
    this.object.putInt("matchLevelNFCToFaceMap", this.defaultIfNull(matchLevelNFCToFaceMap, -1));
    this.object.putInt("faceMapAgeV2GroupEnumInt", this.defaultIfNull(faceMapAgeV2GroupEnumInt, -1));
    this.object.putInt("watermarkAndHologramStatusEnumInt", this.defaultIfNull(watermarkAndHologramStatusEnumInt, -1));
  }

  public WritableMap getMap() {
    return this.object.copy();
  }
}
